lib/index.js

```diff
@@ -49,17 +49,21 @@ class Parser {
     if (typeof tags.timestamp === 'undefined') {
       tags.timestamp = new Date().toISOString()
     }
-
-    let valid = utils.valid(sentence, this.options.validateChecksum)
     
+    //delete signs before $ if $ is availible
+    if (sentence.indexOf("$") !== -1) {
```

Comment:
You should store the result of `indexOf` away here so that you don't do it twice:

```
let idx = sentence.indexOf("$")
if ( idx !== -1 ) {
  ...
}
```

Reply:
You should also check for zero, no need to do the substring if at zero.

```
if ( idx !== -1 && idx !== 0 ) {
...
}
```